__tests__/integration/one-app.spec.js

```diff
@@ -962,6 +953,158 @@ describe('Tests that require Docker setup', () => {
+                    `${appAtTestUrls.cdnUrl}/app/[one-app-version]/vendors.js`,
+                    `${appAtTestUrls.cdnUrl}/app/[one-app-version]/i18n/en-US.js`,
+                    `${appAtTestUrls.cdnUrl}/app/[one-app-version]/app.js`,
+                    `${appAtTestUrls.cdnUrl}/app/[one-app-version]/service-worker-client.js`,
+                  ]
+                )
+              );
+            });
+
+            test('caches a module and its code-split chunk', async () => {
+              expect.assertions(2);
+
+              await browser.url(`${appAtTestUrls.browserUrl}/demo/franks-burgers`);
+              const openerMessage = await browser.$('#franks-opening-line');
+              await openerMessage.waitForExist();
+              const btn = await browser.$('#order-burger-btn');
+              await btn.click();
+              const franksBurger = await browser.$('#franks-burger');
+              await franksBurger.waitForExist();
+
+              const holocronModuleMap = readModuleMap();
+              const cacheKeys = await browser.executeAsync(getCacheKeys);
+              const cacheMap = new Map(await browser.executeAsync(getCacheEntries, cacheKeys));
+
+              expect(cacheMap.get('__sw/modules')).toHaveLength(4);
+              expect(cacheMap.get('__sw/modules')).toEqual(
+                [
+                  holocronModuleMap.modules['frank-lloyd-root'].browser.url,
+                  holocronModuleMap.modules['preview-frank'].browser.url,
+                  holocronModuleMap.modules['franks-burgers'].browser.url,
+                  // the module chunk
+                  holocronModuleMap.modules['franks-burgers'].browser.url.replace(
+                    'franks-burgers.browser.js',
+                    'Burger.franks-burgers.chunk.browser.js'
+                  ),
+                ]
+              );
+            });
+          });
+
+          describe('cache invalidation', () => {
+            beforeAll(async () => {
+              await addModuleToModuleMap({
+                moduleName: 'late-frank',
+                version: '0.0.0',
+              });
+              // wait for change to be picked up
+              await waitFor(5000);
+            });
+
+            describe('module version', () => {
+              const fetchServiceWorkerScript = () => fetch(`${appAtTestUrls.fetchUrl}/_/pwa/service-worker.js`, {
+                ...defaultFetchOptions,
+              }).then((res) => res.text());
+
+              test('service worker script is updated when the module map changes with new module version', async () => {
+                expect.assertions(2);
+                // due to the service worker script changing and being re-installed
+                // selenium does not seem to update the service worker, we can check if the
+                // service worker script is updated to make sure the latest module would be updated
```

Comment:
Any way we may restart selenium or spool a new process of selenium to validate the changed state of service worker?

Reply:
I'm not sure that is what we'd want in this case, since if we'd restart selenium it would clear the cache that needs to be invalidated against. I have tried a few things including `browser.reloadSession` and `browser.refresh` to try getting the browser to. run the invalidation but without success..

Reply:
Guess this is the best that can be done in the current effort. 👍